packages/back-end/src/models/ExperimentModel.ts

```diff
@@ -802,4 +805,54 @@ export const logExperimentDeleted = async (
   return emittedEvent.id;
 };
 
-// endregion Events
+const isValidVisualExperiment = (
+  e: Partial<VisualExperiment>
+): e is VisualExperiment => !!e.experiment && !!e.visualChangeset;
+
+export const getAllVisualExperiments = async (
+  organization: string
+): Promise<Array<VisualExperiment>> => {
+  const visualChangesets = await findVisualChangesets(organization);
+
+  if (!visualChangesets.length) return [];
+
+  const visualChangesByExperimentId = visualChangesets.reduce(
+    (acc: Record<string, Array<VisualChange>>, c) => {
+      if (!acc[c.experiment]) acc[c.experiment] = [];
+      acc[c.experiment] = acc[c.experiment].concat(c.visualChanges);
+      return acc;
+    },
+    {}
+  );
+
+  const experiments = (
+    await findExperiments({
+      id: {
+        $in: visualChangesets.map((changeset) => changeset.experiment),
+      },
+      organization,
+      archived: false,
+    })
+  )
+    // exclude experiments that are stopped and don't have a released variation
+    .filter((e) => e.status !== "stopped" || e.releasedVariationId !== null)
+    // If the experiment is stopped and the released variation doesn’t have any visual changes, it should be excluded
+    .filter(
+      (e) =>
+        e.status !== "stopped" ||
+        visualChangesByExperimentId[e.id].some(
+          (vc) =>
+            vc.variation === e.releasedVariationId &&
+            (!!vc.css || !!vc.domMutations.length)
+        )
+    );
+
+  const visualExperiments: Array<
+    Partial<VisualExperiment>
+  > = visualChangesets.map((c) => ({
+    experiment: experiments.find((e) => e.id === c.experiment),
```

Comment:
Can this potentially return undefined? If so, `isValidVisualExperiment` isn't safely accessing it.

Reply:
I think you're right, I got confused about the object with `experiment` and the visual experiment being a separate object with experiment on it. I think it's fine.